drivers/scheduler/scheduler.go

```diff
@@ -339,8 +342,12 @@ type Driver interface {
 	// CreateCsiSnapshot create csi snapshot for given pvc
 	CreateCsiSnapshot(name string, namespace string, class string, pvc string) (*v1beta1.VolumeSnapshot, error)
 
-	// CreateCsiSnapsAndValidate create csi snapshot and return a pvc using that snapshot
-	CreateCsiSnapsAndValidate(*Context, string) error
+	// CSISnapshotTest create csi snapshot and return a pvc using that snapshot
+	CSISnapshotTest(*Context, CSISnapshotRequest) error
```

Comment:
I am OK for this PR adding the `CSISnapshotTest` and other functions you recently added to the Driver interface. But think again for the nature of Driver interface operation: `CreateCSISnapshot` is better than `CSISnapshotTest`. 
From the SW architecture point of view, you should remove the validation part in the current function. Driver can create CSI snapshot but it shouldn't do the validation.

Reply:
yea I agree there's probably better place to place the validation. Though functions like `ValidatePureVolumesNoReplicaSets` and `ValidateGetByteUsedForVolume` are also under the drivers package (/torpedo/drivers/volume/portworx). Our test is a little special in that its creating separate objects from the "normal" portworx operation, so its also not exactly portworx operation. I can create a separate ticket for improvement if thats needed.

Reply:
Sounds good. Let's add a TODO here.